web-server/sgx-wallet-impl/src/wallet_operations/sign_transaction.rs

```diff
@@ -0,0 +1,126 @@
+        return Result::InvalidAuth;
+    }
+
+    // Authenticated! Proceed to sign the transaction.
+
+    // Safety: `split_at` panics if len < mid.
+    if request.algorand_transaction_bytes.len() < 2 {
+        return Result::Failed(errors::message_with_base64(
+            "sign_transaction",
+            "transaction too short",
+            format!("len = {}", request.algorand_transaction_bytes.len()),
+            "unsigned transaction msgpack",
+            &request.algorand_transaction_bytes,
+        ));
+    }
+
+    // Check and strip the "TX" prefix tag
+    let transaction_bytes_without_prefix = match request.algorand_transaction_bytes.split_at(2) {
+        (b"TX", rest) => rest,
+        (unrecognised, _rest) => {
+            return Result::Failed(errors::message_with_base64(
+                "sign_transaction",
+                "transaction prefix tag not recognised",
+                format!("expected TX, got {:?}", unrecognised),
+                "unsigned transaction msgpack",
+                &request.algorand_transaction_bytes,
+            ))
+        }
+    };
+
+    let transaction_to_sign =
+        &match algorand_network_compatible::from_msgpack(transaction_bytes_without_prefix) {
+            Ok(ok) => ok,
+            Err(err) => {
+                return Result::Failed(errors::message_with_base64(
+                    "sign_transaction",
+                    "failed to unpack received transaction-to-be-signed",
+                    err,
+                    "unsigned transaction msgpack",
+                    &request.algorand_transaction_bytes,
+                ))
+            }
+        };
+
+    let signing_account = &stored.algorand_account.as_algonaut_account();
+
+    let signed_transaction = &match signing_account.sign_transaction(transaction_to_sign) {
+        Ok(ok) => ok,
+        Err(err) => {
+            return Result::Failed(errors::message_with_debug_value(
+                "sign_transaction",
+                "algonaut error while signing transaction",
+                err,
+                "unsigned transaction",
+                transaction_to_sign,
+            ))
+        }
+    };
+
+    // Note: Intentionally serialize with algonaut's `to_msg_pack` helper, not ours.
+    let signed_transaction_bytes = match algorand_network_compatible::to_msgpack(signed_transaction)
+    {
+        Ok(ok) => ok,
+        Err(err) => {
+            return Result::Failed(errors::message_with_debug_value(
+                "sign_transaction",
+                "failed to pack signed ",
+                err,
+                "signed transaction msgpack",
+                signed_transaction,
+            ))
+        }
+    };
+
+    Result::Signed(AlgorandTransactionSigned {
+        signed_transaction_bytes,
+    })
+}
+
+/// Algorand network-compatible MessagePack serialization and deserialization.
+///
+/// This is important to get right: variations will lead to data
```

Comment:
Not sure I understand the sentence.

Reply:
Good catch, thanks! Just incomplete typing on my part: fixed.